File: c/insertion_sort.c
#include <stdio.h>

void print_arr(int arr[], int arr_len) {
    for (int i = 0; i < arr_len; i++)
        printf("%d ", arr[i]);
    printf("\n");
}

void insertion_sort(int arr[], int arr_len) {
    for (int i = 1; i < arr_len; i++) {
        int temp = arr[i];
        int j = i - 1;
        while (j >= 0 && arr[j] > temp) {
            arr[j + 1] = arr[j];
            j--;
        }
        arr[j + 1] = temp;
    }
}

int main() {
    int arr[] = {5, 9, 3, 1, 8, 6, 4, 2, 7};
    int arr_len = sizeof(arr) / sizeof(arr[0]);

    printf("Before: ");
    print_arr(arr, arr_len);

    insertion_sort(arr, arr_len);

    printf("After: ");
    print_arr(arr, arr_len);
    return 0;
}

File: python/insertion_sort.py
def insertion_sort(arr):
    """Function implementing Insertion Sort"""
    arr_len = len(arr)
    for i in range(1, arr_len):
        temp = arr[i]
        j = i - 1
        while j >= 0 and arr[j] > temp:
            arr[j + 1] = arr[j]
            j -= 1
        arr[j + 1] = temp


def main():
    """Driver method"""
    arr = [5, 9, 3, 1, 8, 6, 4, 2, 7]
    print("Before: ", arr)
    insertion_sort(arr)
    print("After: ", arr)


if __name__ == "__main__":
    main()
